plugins/module_utils/network/sonic/config/bgp_neighbors/bgp_neighbors.py

```diff
@@ -797,6 +881,59 @@ def delete_specific_peergroup_param_request(self, vrf_name, cmd):
                             if allowas_in.get('value', None):
                                 delete_path = delete_static_path + '/afi-safis/afi-safi=%s/allow-own-as/config/as-count' % (afi_safi_name)
                                 requests.append({'path': delete_path, 'method': DELETE})
+                        if prefix_list_in:
+                            delete_path = delete_static_path + '/afi-safis/afi-safi=%s/prefix-list/config/import-policy' % (afi_safi_name)
+                            requests.append({'path': delete_path, 'method': DELETE})
+                        if prefix_list_out:
+                            delete_path = delete_static_path + '/afi-safis/afi-safi=%s/prefix-list/config/export-policy' % (afi_safi_name)
+                            requests.append({'path': delete_path, 'method': DELETE})
+                        if afi_safi == 'IPV4_UNICAST':
+                            if ip_afi:
+                                requests.extend(self.delete_ip_afi_requests(ip_afi, afi_safi_name, 'ipv4-unicast', delete_static_path))
+                            if prefix_limit:
+                                requests.extend(self.delete_prefix_limit_requests(prefix_limit, afi_safi_name, 'ipv4-unicast', delete_static_path))
+                        elif afi_safi == 'IPV6_UNICAST':
+                            if ip_afi:
+                                requests.extend(self.delete_ip_afi_requests(ip_afi, afi_safi_name, 'ipv6-unicast', delete_static_path))
+                            if prefix_limit:
+                                requests.extend(self.delete_prefix_limit_requests(prefix_limit, afi_safi_name, 'ipv6-unicast', delete_static_path))
+                        elif afi_safi == 'L2VPN_EVPN':
+                            if prefix_limit:
+                                requests.extend(self.delete_prefix_limit_requests(prefix_limit, afi_safi_name, 'l2vpn-evpn', delete_static_path))
+
+        return requests
+
+    def delete_ip_afi_requests(self, ip_afi, afi_safi_name, afi_safi, delete_static_path):
+        requests = []
+        default_policy_name = ip_afi.get('default_policy_name', None)
+        send_default_route = ip_afi.get('send_default_route', None)
+        if default_policy_name:
+            delete_path = delete_static_path + '/afi-safis/afi-safi=%s/%s/config/default-policy-name' % (afi_safi_name, afi_safi)
+            requests.append({'path': delete_path, 'method': DELETE})
+        if send_default_route:
+            delete_path = delete_static_path + '/afi-safis/afi-safi=%s/%s/config/send_default_route' % (afi_safi_name, afi_safi)
+            requests.append({'path': delete_path, 'method': DELETE})
+
```

Comment:
Don't we need handling for bulk deletion of the whole ip_afi "group"? (This same question applies to "delete_prefix_limit_requests".)

Reply:
Further investigation into this question has shown that there is no good way, with the current Ansible infrastructure, to use a YAML playbook to request deletion of a dictionary that is not a list item. This is because there is no way to differentiate, in the command list returned by the "AnsibleModule" method, between a dictionary that is not specified at all in the input playbook versus a dictionary for which only the name is specified in the playbook: Both cases yield "'dict_name': None" with the yaml.safe_load() method and a "null" value in the parsed input command list.

In addition, currently existing resource module implementations are not providing the ability to delete a dictionary that is not a list item.

For these reasons, I am dropping this request.